src/models/vamp.jl

```diff
@@ -0,0 +1,96 @@
+export VAMP, mmd_mean_vamp, init_vamp_mean, init_vamp_sample
+import ConditionalDists: rand # so that there is no conflict
+"""
+	VAMP{K<:Int,P<:AbstractArray}(K::Int, xdim::Union{Tuple, Int})
+	VAMP{K<:Int,P<:AbstractArray}(X::P)
+
+Vamp prior with `K` pseudoinputs `P`. If only the pseudoinput `X` is given, `K` is the size of 
+the last dimension. 
+
+# Arguments
+* `K`: Number of pseudoinputs
+* `xdim`: Size of an individual pseudoinput
+* `X`: Pseudoinput array
+
+# Example
+Initialize a VAMP with K pseudoinputs of size xdim:
+```julia-repl
+julia> v = VAMP(2, 2)
+VAMP{Int64,Array{Float32,2}}(
+K: 2
+pseudoinputs: Float32[0.876433 0.56233144; 0.24732344 0.44447201]
+)
+
+julia> v = VAMP(2, (3, 2))
+VAMP{Int64,Array{Float32,3}}(
+K: 2
+pseudoinputs: Float32[0.21458945 0.8710681; 0.2402707 -0.20706704; 0.9916858 0.6571086]
+
+Float32[0.6324719 0.27955383; -0.39046887 0.18451884; -0.18170312 -1.3017029]
```

Comment:
is that an overloaded show function? or does julia do that automatically by now?

Reply:
This is automatically generated by Julia, maybe a custom show function would be nice.